### PR TITLE
Addbackprop

### PR DESCRIPTION
**Pull request by smileandyxu**
Add backpropagation function.

### CODE REVIEW

**Comment by smileandyxu** [comment]
Finished:

View: provides backpropagation related button in menu.
ViewModel&Model: realize gradient descent backpropagation algorithm in three steps, each can be perform individually.
